### DISCUSSION TITLE
how to used server import.meta.url ?

### DISCUSSION THREAD

**Discussion by productdevbook** [nuxt/nuxt#17730] [Questions]
iam used server

import.meta.url = undefined

maybe used __dirname but is not defined in ES module scope

**Comment by productdevbook** [comment]
const runtimeDir = fileURLToPath(new URL('.', import.meta.url))

web:dev: [nitro] [dev] [uncaughtException] TypeError [ERR_INVALID_URL]: Invalid URL

**Comment by pi0** [comment]
Hi. This is intentional and you shouldn't depend on the filesystem for server routes. What is your usecase?

**Reply by productdevbook** [reply]
![image](https://user-images.githubusercontent.com/38668796/162737054-4b30ade7-7ee0-47af-8145-2611f3a33a19.png)

**Reply by productdevbook** [reply]
how do i have to do this, is this nuxt server possible?

**Comment by awdr74100** [comment]
Hi @productdevbook @pi0 Is there any solution? `fileURLToPath(new URL('./resolvers', import.meta.url))` is `/Users/xxx/Desktop/demo/.nuxt/dev/resolvers`. Expected should be `/Users/xxx/Desktop/demo/server/resolvers`.
